critwall/src/main/java/org/opencadc/critwall/FileSyncJob.java

```diff
@@ -0,0 +1,212 @@
+import java.io.ByteArrayOutputStream;
+import java.io.FileNotFoundException;
+import java.net.URI;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.log4j.Logger;
+import org.opencadc.inventory.Artifact;
+import org.opencadc.inventory.InventoryUtil;
+import org.opencadc.inventory.db.ArtifactDAO;
+import org.opencadc.inventory.storage.NewArtifact;
+import org.opencadc.inventory.storage.StorageAdapter;
+
+import org.opencadc.inventory.storage.StorageMetadata;
+
+public class FileSyncJob implements Runnable {
+    private static final Logger log = Logger.getLogger(FileSyncJob.class);
+
+    private final ArtifactDAO artifactDAO;
+    private final URI artifactID;
+    private final URI resourceID;
+    private final StorageAdapter storageAdapter;
+
+    public FileSyncJob(URI artifactID, URI resourceID, StorageAdapter storageAdapter, ArtifactDAO artifactDAO) {
+        InventoryUtil.assertNotNull(FileSyncJob.class, "artifactID", artifactID);
+        InventoryUtil.assertNotNull(FileSyncJob.class, "resourceID", resourceID);
+        InventoryUtil.assertNotNull(FileSyncJob.class, "storageAdapter", storageAdapter);
+        InventoryUtil.assertNotNull(FileSyncJob.class, "artifactDAO", artifactDAO);
+
+        this.artifactID = artifactID;
+        this.resourceID = resourceID;
+        this.storageAdapter = storageAdapter;
+        this.artifactDAO = artifactDAO;
+    }
+
+    @Override
+    public void run() {
+        RegistryClient regClient = new RegistryClient();
+        log.debug("resource id: " + this.resourceID);
+        URL certURL = regClient.getServiceURL(this.resourceID, Standards.VOSPACE_SYNC_21, AuthMethod.CERT);
+        log.debug("certURL: " + certURL);
+
+        try {
+            // Ask for all protocols available back, and the server will
+            // give you URLs for the ones it allows.
+            List<Protocol> protocolList = new ArrayList<Protocol>();
+            Protocol httpsCert = new Protocol(VOS.PROTOCOL_HTTPS_GET);
+
+            log.debug("protocols: " + httpsCert);
+            protocolList.add(httpsCert);
+
+            Transfer transfer = new Transfer(artifactID, Direction.pullFromVoSpace, protocolList);
+            transfer.version = VOS.VOSPACE_21;
+
+            TransferWriter writer = new TransferWriter();
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            writer.write(transfer, out);
+            FileContent content = new FileContent(out.toByteArray(), "text/xml");
+            log.debug("xml file content to be posted: " + transfer);
+
+            log.debug("artifact path: " + artifactID.getPath());
+            HttpPost post = new HttpPost(certURL, content, true);
+            post.prepare();
+            log.debug("post prepare done");
+
+            // TODO: error handling is likely very different than this...
+            if (post.getThrowable() != null && post.getThrowable() instanceof FileNotFoundException) {
```

Comment:
post.getThrowable() is null if you call prepare() because prepare() throws exceptions instead of capturing them; this can be removed.

Reply:
done